Add and split documentation into basic and advanced use

timtroendle opened the pull request:
* Fix #154
* Split documentation into basic and advanced use (pre-builts vs workflow)
* Move all documentation to doc folder
* Restructure documentation
* Document all customisation options
* Move from Sphinx to mkdocs

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

brynpickering commented on docs/index.md:
```diff
@@ -0,0 +1,17 @@
+# Euro Calliope
+
+Models of the European electricity system for Calliope.
```
`for the Calliope energy system modelling framework`?

brynpickering commented on docs/index.md:
```diff
@@ -0,0 +1,17 @@
+
+![](./img/ec-in-modelling-process.png)
+
+## At a glance
```
Figures here would be nice.

brynpickering commented on docs/sources.md:
```diff
@@ -1,47 +1,47 @@
 European Environment Agency. (2009). <i>Europe’s onshore and offshore wind energy potential</i> [Publication]. https://www.eea.europa.eu/publications/europes-onshore-and-offshore-wind-energy-potential
 
-> [@Ruiz Catello et al:2015]
+### @Ruiz Catello et al:2015
```
should just be Ruiz?

brynpickering commented on templates/README.md:
```diff
@@ -2,114 +2,35 @@
 Ready to use models of the European electricity system built using Calliope. Models are available on three different spatial resolutions: continental, national, and regional.
 
-In addition, euro-calliope models can be built manually which adds more configuration options. To build euro-calliope manually, head over to [GitHub](https://github.com/calliope-project/euro-calliope).
+In addition, euro-calliope models can be built manually which adds more configuration options. To learn how to build euro-calliope manually, head over to [euro-calliope's documentation](https://euro-calliope.readthedocs.io).
```
I would keep the GH link and mention that it was built from raw data using this workflow, with a pointer to the repo with appropriate tag.

timtroendle replied:
Yeah. But from there users would be pointed back to the documentation anyway. Plus, it's one more link for which we must maintain the correct version. I am already worried that we will mess this up.

brynpickering commented on templates/README.md:
```diff
@@ -2,114 +2,35 @@
 ## At a glance
 
-euro-calliope models the European electricity system with each location representing an administrative unit. It is built on three spatial resolutions: on the continental level as a single location, on the national level with 34 locations, and on the regional level with 497 locations. On each node, renewable generation capacities (wind, solar, bioenergy) and balancing capacities (battery, hydrogen ...
+euro-calliope models the European electricity system with each location representing an administrative unit. It is built on three spatial resolutions: on the continental level as a single location, on the national level with 34 locations, and on the regional level with 497 locations. At each location, renewable generation capacities (wind, solar, bioenergy) and balancing capacities (battery, hydr ...
```
This is the third instance of this paragraph, which might be a pain to deal with in the long run. Any way to depend on one instance of this description to fill in the rest...?

timtroendle replied:
True. I tried to keep text duplicates to a minimum but found this paragraph important enough to copy it. If you have a technical solution, I am happy to hear about it. Otherwise we simply need to maintain the copies.

brynpickering commented on templates/README.md:
I would have the hyperlink also covering 'documentation'

brynpickering reviewed (commented):
Nice rapid implementation.

I'm not keen on the structure though. `basic` and `advanced` are unnecessarily abstract. I would stick with more explicit sections like `Running the pre-built model`, `Customising the pre-built model`, `Building the model from raw data`, `Customising the model building workflow`. `Customising` could be a subset of the `running` and `building` sections, e.g.
```
Running the pre-built model
-- Prepare
---- The models
-- Run
-- Customise
Building the model from raw data
-- Prepare
---- The repository
-- Build
---- Building on a cluster
-- Run (maybe refer to the run phase from the other section)
-- Customise
```

Other general comments:
1. We need more pictures! We could have the network and region image in the index when describing what euro-calliope is and should have the rule graph in the `building the model` section.

2. We should have a `troubleshooting` section for running the euro-calliope model and building it. I guess this fits with the FAQs in #159. Ideally we want to use the documentation not as just a quick-start, but as the first line of defence for being inundated with support requests! This could come in on another PR. 

3. For general maintainability via GH PRs, it would be good to have one sentence per line. Then changes made to the documentation in future can be tracked a lot more easily. Provided there isn't an empty line, it will be rendered as one paragraph in the end.

timtroendle commented:
I have addressed most of your comments.

The ones I haven't addressed are parts of other issues: network graph (#165), DAG visualisation (#108), troubleshooting (#159).

brynpickering commented on mkdocs.yaml:
```diff
@@ -0,0 +1,23 @@
+nav:
+    - Home: index.md
+    - Access:
+        - Use pre-builts: model/pre-built.md
```
```suggestion
        - Use pre-built models: model/pre-built.md
```

brynpickering reviewed (approved):
One minor change request, otherwise looks great!
